diff --git a/node_modules/@dataplan/pg/dist/steps/pgSelect.js b/node_modules/@dataplan/pg/dist/steps/pgSelect.js
index c61e26e..3bf7abf 100644
--- a/node_modules/@dataplan/pg/dist/steps/pgSelect.js
+++ b/node_modules/@dataplan/pg/dist/steps/pgSelect.js
@@ -1057,9 +1057,6 @@ and ${pg_sql2_1.default.indent(pg_sql2_1.default.parens(condition(i + 1)))}`}
         };
     }
     buildQuery(options = Object.create(null)) {
-        if (!this.isTrusted) {
-            this.resource.applyAuthorizationChecksToPlan(this);
-        }
         const reverse = this.shouldReverseOrder();
         const { sql: select, extraSelectIndexes } = this.buildSelect(options);
         const { sql: from } = this.buildFrom();
@@ -1303,6 +1300,10 @@ ${lateralText};`;
         super.finalize();
     }
     deduplicate(peers) {
+        if (!this.isTrusted) {
+            this.resource.applyAuthorizationChecksToPlan(this);
+            this.isTrusted = true;
+        }
         this.locker.lockAllParameters();
         return peers.filter(($p) => {
             if ($p === this) {
